src/data/list/nodup.lean

```diff
@@ -322,6 +322,22 @@ begin
   { simp [ne.symm H, H, update_nth, ← apply_ite (cons (f hd))] }
 end
 
+lemma nodup.pairwise_of_forall_ne {l : list α} {r : α → α → Prop}
+  (hl : l.nodup) (h : ∀ (a ∈ l) (b ∈ l), a ≠ b → r a b) : l.pairwise r :=
```

Comment:
Does it make sense to have this variant?
```suggestion
lemma nodup.pairwise_of_forall_ne {l : list α} {r : α → α → Prop}
  (hl : l.nodup) (h : set.pairwise_on {a | a ∈ l} r) : l.pairwise r :=
```

Do we have a better way of spelling `{a | a ∈ l}` already? Is there a coercion from list to set?

Reply:
We don't have coercions from lists to set, unless you consider `list` > `finset` > `set`, which requires `decidable_eq`. I'll add a lemma converting form `list.pairwise` to `set.pairwise_on`. I think their combination is unnecessary.